core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2880,8 +2890,7 @@ __wasi_errno_t
 wasmtime_ssp_sched_yield(void)
 {
 #ifdef BH_PLATFORM_WINDOWS
-    if (!SwitchToThread())
-        return __WASI_EAGAIN;
+    SwitchToThread();
```

Comment:
For this change, I'm not sure if we should fail if it doesn't succeed - that implies there are no other threads to switch to which isn't really a failure (https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-switchtothread). Also, [`sched_yield`](https://pubs.opengroup.org/onlinepubs/007904975/functions/sched_yield.html) doesn't define any error codes so I'm not sure callees expect this function call to ever fail.

Reply:
That looks good to me. Regarding posix's `shed_yield`, I'd keep it as it is. The documentation clearly states that `-1` means error, even though no `errno` code is specified.